src/emit/binary.rs

```diff
@@ -829,7 +829,13 @@ impl<'a> Binary<'a> {
                         &ns.structs[*n]
                             .fields
                             .iter()
-                            .map(|f| self.llvm_field_ty(&f.ty, ns))
+                            .map(|f| {
+                                if f.ty.is_mapping() {
+                                    self.context.i32_type().into()
```

Comment:
On  Substrate/ewasm, it's not i32, but i256. 

`bin.llvm_type(&ns.storage_type(), ns).into_int_type()` ?